flake.nix

```diff
@@ -0,0 +1,191 @@
+
+    flake-utils = {
+      url = "github:numtide/flake-utils";
+    };
+
+    flake-compat = {
+      url = "github:edolstra/flake-compat";
+      flake = false;
+    };
+
+    rust-overlay = {
+      url = "github:oxalica/rust-overlay";
+      # All of these inputs (a.k.a. dependencies) need to align with inputs we
+      # use so they use the `inputs.*.follows` syntax to reference our inputs
+      inputs = {
+        nixpkgs.follows = "nixpkgs";
+        flake-utils.follows = "flake-utils";
+      };
+    };
+
+    crane = {
+      url = "github:ipetkov/crane";
+      # All of these inputs (a.k.a. dependencies) need to align with inputs we
+      # use so they use the `inputs.*.follows` syntax to reference our inputs
+      inputs = {
+        nixpkgs.follows = "nixpkgs";
+        flake-utils.follows = "flake-utils";
+        flake-compat.follows = "flake-compat";
+        rust-overlay.follows = "rust-overlay";
+      };
+    };
+
+    barretenberg = {
+      url = "github:AztecProtocol/barretenberg";
+      # All of these inputs (a.k.a. dependencies) need to align with inputs we
+      # use so they use the `inputs.*.follows` syntax to reference our inputs
+      inputs = {
+        nixpkgs.follows = "nixpkgs";
+        flake-utils.follows = "flake-utils";
+      };
+    };
+  };
+
+  outputs =
+    { self, nixpkgs, crane, flake-utils, rust-overlay, barretenberg, ... }:
+    flake-utils.lib.eachDefaultSystem (system:
+    let
+      pkgs = import nixpkgs {
+        inherit system;
+        overlays = [
+          rust-overlay.overlays.default
+          barretenberg.overlays.default
+        ];
+      };
+
+      rustToolchain = pkgs.rust-bin.stable."1.66.0".default.override {
+        # We include rust-src to ensure rust-analyzer works.
+        # See https://discourse.nixos.org/t/rust-src-not-found-and-other-misadventures-of-developing-rust-on-nixos/11570/4
+        extensions = [ "rust-src" ];
+      };
+
+      craneLib = (crane.mkLib pkgs).overrideToolchain rustToolchain;
+
+      environment = {
+        # rust-bindgen needs to know the location of libclang
+        LIBCLANG_PATH = "${pkgs.llvmPackages.libclang.lib}/lib";
+
+        # Barretenberg fails if tests are run on multiple threads, so we set the test thread
+        # count to 1 throughout the entire project
+        #
+        # Note: Setting this allows for consistent behavior across build and shells, but is mostly
+        # hidden from the developer - i.e. when they see the command being run via `nix flake check`
+        RUST_TEST_THREADS = "1";
+
+        # We set the environment variable because barretenberg must be compiled in a special way for wasm
+        BARRETENBERG_BIN_DIR = "${pkgs.barretenberg-wasm}/bin";
+
+        # We provide `barretenberg-transcript00` from the overlay to the build.
+        # This is necessary because the Nix sandbox is read-only and downloading during tests would fail
+        BARRETENBERG_TRANSCRIPT = pkgs.barretenberg-transcript00;
```

Comment:
How does this work for larger circuits whom may need more points?

Reply:
I think our backend only currently downloads transcript00 at the moment anyway. Perhaps we should add a new issue to handle larger circuits that need more points and thus a larger transcript

Reply:
This is something that @kobyhallx is currently thinking about btw.

Reply:
Actually... can we not remove this @phated? This seems like it's only relevant in the flake for `aztec_backend`.

Reply:
Do we not inherit from when it's set in `aztec_backend` though? That said, we also need to specify `barretenberg` even though that's only used in `aztec_backend` too.

In any case it's not a blocker and I can experiment later.

Reply:
Unfortunately (or maybe fortunately ⁉️ ), we don't inherit anything from flakes in any rust projects because they are consumed as crates, not flakes.